crypto/randentropy/rand_entropy.go

```diff
@@ -0,0 +1,82 @@
+package randentropy
+
+import (
+	crand "crypto/rand"
+	"encoding/binary"
+	"github.com/ethereum/go-ethereum/crypto/sha3"
+	"io"
+	"os"
+	"strings"
+	"time"
+)
+
+type RandEntropy struct {
```

Comment:
I suggest we make this an unexported type and export `randentropy.Reader`and remove the `new(randentropy.RandEntropy)` from the other files and pass in the `Reader` directly.